DQM/CTPPS/plugins/CTPPSDiamondDQMSource.cc

```diff
@@ -0,0 +1,854 @@
+      if ( ! track.isValid() ) continue;
+      if ( excludeMultipleHits_ && track.getMultipleHits() > 0 ) continue;
+      if ( potPlots_.find( detId_pot ) == potPlots_.end() ) continue;
+
+      const double rnd = ( 2. * randomGen.Rndm() ) - 1.;
+      if ( track.getOOTIndex() == 1 ) potPlots_[detId_pot].trackDistribution->Fill( track.getX0()+rnd*track.getX0Sigma() );
+      potPlots_[detId_pot].trackDistributionOOT->Fill( track.getOOTIndex(), track.getX0()+rnd*track.getX0Sigma() );
+    }
+  }
+  
+  // Tomography of diamonds using strips
+  for ( const auto& rechits : *diamondRecHits ) {
+    CTPPSDiamondDetId detId_pot( rechits.detId() );
+    detId_pot.setPlane( 0 );
+    detId_pot.setChannel( 0 );
+    const CTPPSDiamondDetId detId( rechits.detId() );
+
+    for ( const auto& rechit : rechits ) {
+      if ( excludeMultipleHits_ && rechit.getMultipleHits() > 0 ) continue;
+      if ( !stripTracks.isValid() ) continue;
+      if ( potPlots_.find( detId_pot ) == potPlots_.end() ) continue;
+
+      for ( const auto& ds : *stripTracks ) {
+        const CTPPSDetId stripId( ds.detId() );
+        for ( const auto& striplt : ds ) {
+          if ( !striplt.isValid() ) continue;
+          if ( stripId.arm() != detId_pot.arm() ) continue;
+          if ( striplt.getTx() > minimumStripAngleForTomography_ || striplt.getTy() > minimumStripAngleForTomography_) continue; 
+          if ( stripId.rp() == 3 ) {
+            switch ( rechit.getOOTIndex() ) {
+              case 0: {
+                potPlots_[detId_pot].stripTomographyAllFar_minus1->Fill( striplt.getX0() + 50*detId.plane(), striplt.getY0() );
+              } break;
+              case 1: {
+                potPlots_[detId_pot].stripTomographyAllFar->Fill( striplt.getX0() + 50*detId.plane(), striplt.getY0() );
+              } break;
+              case 2: {
+                potPlots_[detId_pot].stripTomographyAllFar_plus1->Fill( striplt.getX0() + 50*detId.plane(), striplt.getY0() );
+              } break;
+            }
+          }
+          else if ( stripId.rp() == 2 ) {
+            switch ( rechit.getOOTIndex() ) {
+              case 0: {
+                potPlots_[detId_pot].stripTomographyAllNear_minus1->Fill( striplt.getX0() + 50*detId.plane(), striplt.getY0() );
+              } break;
+              case 1: {
+                potPlots_[detId_pot].stripTomographyAllNear->Fill( striplt.getX0() + 50*detId.plane(), striplt.getY0() );
+              } break;
+              case 2: {
+                potPlots_[detId_pot].stripTomographyAllNear_plus1->Fill( striplt.getX0() + 50*detId.plane(), striplt.getY0() );
+              } break;
+            }
+          }
+        }
+      }
+    }
+  }
+
+  //------------------------------
+  // Clock Plots
+  //------------------------------
+
+  for ( const auto& digis : *diamondDigis ) {
+    const CTPPSDiamondDetId detId( digis.detId() );
+    CTPPSDiamondDetId detId_pot( digis.detId() );
+    if ( detId.channel() == 30 ) {
```

Comment:
hi @nminafra - thanks - in a followup PR, please define a const variable for "30" so that its meaning is more clear (and besides, things that never change have a tendency to change:) ).